### PR TITLE
[New Sample] - Bot-Configuration Node Js

### PR DESCRIPTION
**Pull request by Harikrishnan-MSFT**

### CODE REVIEW

**Comment by Harikrishnan-MSFT** [comment]
**Build Pipeline -**

**Comment by ChetanSharma-msft** [comment]
image

File: README.md [hunk 1]
```diff
@@ -83,6 +83,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |20| Different formatting on cards | This sample feature shows how to use different formatting on cards using bot. |[View][BotFormattingCards#csharp]|[View][BotFormattingCards#nodejs] ![toolkit-icon](assets/toolkit-icon.png) 
 |21| Different types of cards using bot | This sample shows the feature where user can interact with different types of cards using bot like Adaptive, hero, list, Office 365 Connector, receipt, sign in, and thumbnail cards and card collections.|[View][TypesCards#csharp] | [View][TypesCards#nodejs] ![toolkit-icon](assets/toolkit-icon.png) 
 |22| User specific views in Adaptive cards | This sample shows the different ways developers can consume user specific views in Adaptive cards. |[View][botadaptivecardsuserspecificviews#csharp]| | | |[View](/samples/bot-adaptivecards-user-specific-views/csharp/demo-manifest/bot-adaptivecards-user-specific-views.zip)
+|23| Bot configuration app with type ahead search control.|Demonstrating the feature of configurable card with type ahead search (static and dynamic) control on Adaptive Cards.||[View][BotConfigurationApp#nodejs]
```
Comment (ChetanSharma-msft) [inline README.md:86]: This is not talking about anything on bot configuration.

File: samples/bot-configuration-app/nodejs/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  ![configurable-card-Interaction3 ](Images/configurable-card-Interaction3.png)
+    
+  ![configurable-card-Interaction4 ](Images/configurable-card-Interaction4.png)
+
+   - **Bot description card Interactions:**
+
+   **Bot Descrption Card:**
+  ![Bot-description-card](Images/Bot-description-card.png)
+
+   **Click the settings button in the card to invoke configurable card:**
```
Comment (ChetanSharma-msft) [inline samples/bot-configuration-app/nodejs/README.md:142]: Hover on bot and open configuration popup step is missing here.

File: samples/bot-configuration-app/nodejs/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+> Note: Provisioning and deployment may incur charges to your Azure Subscription.  
+
+## Further reading
+
+- [Bot Framework Documentation](https://docs.botframework.com)
+- [Bot Basics](https://docs.microsoft.com/azure/bot-service/bot-builder-basics?view=azure-bot-service-4.0)
+- [Send Notification to User in Chat](https://docs.microsoft.com/en-us/graph/api/chat-sendactivitynotification?view=graph-rest-beta)
+- [Send Notification to User in Team](https://docs.microsoft.com/en-us/graph/api/team-sendactivitynotification?view=graph-rest-beta&tabs=http)
+- [Send Notification to User](https://docs.microsoft.com/en-us/graph/api/userteamwork-sendactivitynotification?view=graph-rest-beta&tabs=http)
```
Comment (ChetanSharma-msft) [inline samples/bot-configuration-app/nodejs/README.md:167]: Bot  config doc is not linked.

File: samples/bot-configuration-app/nodejs/appManifest/manifest.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          }
+        },
+        "isNotificationOnly": false
+      }
+    ],
+    "permissions": [ "identity", "messageTeamMembers" ],
+    "validDomains": [
+      "${{BOT_DOMAIN}}",
+      "teams.live.com",
+      "teams.microsoft.com",
```
Comment (ChetanSharma-msft) [inline samples/bot-configuration-app/nodejs/appManifest/manifest.json:50]: Remove this.

File: samples/bot-configuration-app/nodejs/assets/sample.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  {
+    "name": "officedev-microsoft-teams-samples-bot-configuration-app-nodejs",
+    "source": "officeDev",
+    "title": "Bot configuration app with type ahead search control",
+    "shortDescription": "This sample shows the feature of bot configuration and reconfiguration for bots in team and group chat.",
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-configuration-app/nodejs",
+    "longDescription": [
+      "This sample demonstrates the features of bot configuration and reconfiguration for both teams and group chats, including a type-ahead search (static and dynamic) control on Adaptive Cards."
+    ],
+    "creationDateTime": "2023-08-23",
```
Comment (ChetanSharma-msft) [inline samples/bot-configuration-app/nodejs/assets/sample.json:11]: Use latest dates. in Readme file as well.

File: samples/bot-configuration-app/nodejs/assets/sample.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        "key": "TEAMS-SERVER-LANGUAGE",
+        "value": "javascript"
+      },
+      {
+        "key": "TEAMS-SERVER-PLATFORM",
+        "value": "express"
+      },
+      {
+        "key": "TEAMS-FEATURES",
+        "value": "bot"
```
Comment (ChetanSharma-msft) [inline samples/bot-configuration-app/nodejs/assets/sample.json:31]: Adaptive card

File: samples/bot-configuration-app/nodejs/package.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    "dependencies": {
+        "@microsoft/microsoft-graph-client": "^2.2.1",
+        "axios": "^0.21.0",
+        "botbuilder": "~4.11.0",
+        "botbuilder-dialogs": "^4.11.2",
+        "cors": "^2.8.5",
+        "dotenv": "^8.2.0",
+        "ejs": "^3.1.6",
+        "express": "^4.17.1",
```
Comment (ChetanSharma-msft) [inline samples/bot-configuration-app/nodejs/package.json:30]: Check if we are not using.

File: README.md (hunk 1 above)
Comment (ChetanSharma-msft) [reply README.md:86]: Run build pipeline.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

**Review by Prithvi-MSFT — APPROVED** [review]
approved

File: README.md (hunk 1 above)
Comment (Jegadeesh-MSFT) [reply README.md]: Fixed

File: samples/bot-configuration-app/nodejs/README.md (hunk 3 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-configuration-app/nodejs/README.md:169]: Fixed

File: samples/bot-configuration-app/nodejs/appManifest/manifest.json (hunk 4 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-configuration-app/nodejs/appManifest/manifest.json]: Fixed

File: samples/bot-configuration-app/nodejs/assets/sample.json (hunk 5 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-configuration-app/nodejs/assets/sample.json]: Fixed

**Comment by Harikrishnan-MSFT** [comment]
**Build Pipeline -**
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6049400439

image

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Pull request merged**